Networking tests are using Go 1.5

I tried to bump our min requirements in the build and the networking extended tests failed.  We don't work against Go 1.5 anymore and need to ensure we get Go updated properly.

```
[INFO] Targeting subnet plugin: redhat/openshift-ovs-subnet
[INFO] Launching a docker-in-docker cluster for the subnet plugin
Cleaning up docker-in-docker containers
Cleanup up configuration to avoid conflict with a future cluster
Cleaning up volumes used by docker-in-docker daemons
Configured network plugin: redhat/openshift-ovs-subnet
Ensuring compatible host configuration
Launching containers
Provisioning nettest-master
Adding '172.17.0.3	nettest-node-1' to hosts file
Adding '172.17.0.4	nettest-node-2' to hosts file
Building openshift

Detected Go version: go version go1.5.4 linux/amd64.
Origin builds require Go version 1.6 or greater.

[ERROR] PID 164: hack/common.sh:258: `( os::build::internal::build_binaries "${binaries[@]+"${binaries[@]}"}" )` exited with status 2.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/common.sh:258: `( os::build::internal::build_binaries "${binaries[@]+"${binaries[@]}"}" )`
[INFO] 		  2: hack/build-go.sh:31: os::build::build_binaries
[INFO]   Exiting with code 2.
[INFO] Saving cluster configuration
```

--- a/images/dind/Dockerfile
+++ b/images/dind/Dockerfile
@@ -4,7 +4,7 @@
 #
 # The standard name for this image is openshift/dind
 #
-FROM fedora:23
+FROM fedora:24
 
 ## Configure systemd to run in a container
 ENV container=docker
@@ -32,7 +32,7 @@ VOLUME ["/run", "/tmp"]
 
 ## Install packages
 RUN dnf -y update && dnf -y install git golang hg tar make findutils \
-  hostname bind-utils iproute iputils which procps-ng openssh-server \
+  gcc hostname bind-utils iproute iputils which procps-ng openssh-server \
   # Node-specific packages
   docker openvswitch bridge-utils ethtool iptables-services \
   && dnf clean all

